docs/php-api.md

```diff
@@ -381,6 +382,176 @@ function my_plugin_register_send_email_ability() {
 }
 ```
 
+#### Registering an Ability with a Custom Ability Class
+
+The `ability_class` parameter allows you to use a custom class that extends `WP_Ability`. This is useful when you need to add custom methods to your ability or override the default behavior of the base `WP_Ability` class.
+
+**When to use a custom ability class:**
+- You need to add custom helper methods specific to your ability
+- You want to override how the ability executes (e.g., add logging, caching, or custom validation)
+- You need to store additional state or configuration beyond what the standard metadata provides
+- You want to encapsulate complex business logic within the ability class itself
+
+**Example: Creating a custom ability class with additional methods**
+
+```php
+/**
+ * Custom ability class that adds a validation helper method.
+ *
+ * This example shows how to extend WP_Ability to add custom behavior
+ * while still leveraging all the standard ability functionality.
+ */
+class My_Plugin_Post_Validator_Ability extends WP_Ability {
+
+    /**
+     * Custom method to check if a post ID is valid and accessible.
+     *
+     * This helper method can be called from the execute callback or permission callback
+     * to perform validation logic that's specific to this ability.
+     *
+     * @param int $post_id The post ID to validate.
+     * @return true|WP_Error True if valid, WP_Error if invalid.
+     */
+    public function validate_post_id( $post_id ) {
```

Comment:
This example seems counterintuitve at a glance. I wanna give it some more thought in the morning but this either should be in the permission callback or in our overloaded `do_execute()`. If we want to give an example of a public method being called on the instance inside of inside our extended class, we might want to think of a use case that better justifies it, maybe something stateful?